app/services/concerns/payloadable.rb

```diff
@@ -14,6 +14,6 @@ def serialize_payload(record, serializer, options = {})
   end
 
   def signing_enabled?
-    true
+    !Rails.configuration.x.whitelist_mode
```

Comment:
This should also be done for `AUTHORIZED_FETCH`, see https://github.com/tootsuite/mastodon/pull/11295